ReactWindows/ReactNative/Accessibility/DynamicAutomationPeer.cs

```diff
@@ -30,7 +30,10 @@ public DynamicAutomationPeer(T owner)
         /// <inheritdoc />
         protected override AutomationControlType GetAutomationControlTypeCore()
         {
-            if (Owner.AccessibilityTraits?.Contains(AccessibilityTrait.Button) == true)
+            if (Owner.AccessibilityTraits?.Contains(AccessibilityTrait.ListItem) == true)
+            {
+                return AutomationControlType.ListItem;
```

Comment:
@zholobov@gmail.com MSDN Outlines what ListViewItemAutomationPeer returns here: [https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.listviewitemautomationpeer](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.listviewitemautomationpeer).
 Most of the functionality is delegated to the base class, including PatternCore (this is why below I return base pattern core). From the rest of the functionality, I think we need nothing else and functionality is in place. Do you have a ref to UIA specs you are referring to?

---
In reply to: [184140661](https://github.com/Microsoft/react-native-windows/pull/1803#discussion_r184140661) [](ancestors = 184140661)

Reply:
This is spec for ListItem control type: [https://msdn.microsoft.com/en-us/library/windows/desktop/ee671635?f=255&MSPPError=-2147217396](https://msdn.microsoft.com/en-us/library/windows/desktop/ee671635?f=255&MSPPError=-2147217396)

I think SelectionItem pattern must be implemented since at least conversation list items are selectable.

Also I do not see implementation of PositionInSet and SizeOfSet UIA properties, so it looks like "x of y" suffix will not be read by narrator.

I also assume VirtualizedITem pattern should be implemented since many list are virtualized, so user must be able to operate them using narrator only. [https://msdn.microsoft.com/en-us/library/windows/desktop/ee671294%28v=vs.85%29.aspx?f=255&MSPPError=-2147217396](https://msdn.microsoft.com/en-us/library/windows/desktop/ee671294%2528v=vs.85%2529.aspx?f=255&MSPPError=-2147217396)

---
In reply to: [184221383](https://github.com/Microsoft/react-native-windows/pull/1803#discussion_r184221383) [](ancestors = 184221383,184140661)

Reply:
@zholobov@gmail.com 
To the points:
1. MSDN describes what the implementation of the ListViewItemAutomationPeer does. It says the pattern is delegated to the base class. Specifically for the selection-related, it says it is supported by the parent list view and not by the ListViewItem. As far as I can say, our implementation of ListView doesn't select the items but sets the focus on these (where the UWP ListView allows for focused and selected items to differ). 
2. "Number x out of y" works. I guess this is because the narrator is has the logic. The method that you refer to, I guess can be use to change the default order (there is an issue with our order of the listitems but I think it is being addressed in the application code)
3. This to me seems like a stretch. My take on this would be to implemented the basics and then if needed we can add to it since a provider is not required to implement all of traits. To keep track if this, I'll include a ref to the msdn doc that describes the traits. 

---
In reply to: [184237387](https://github.com/Microsoft/react-native-windows/pull/1803#discussion_r184237387) [](ancestors = 184237387,184221383,184140661)

Reply:
1. If you look in inspect.exe standard UWP ListViewItem exposes SelectionItem, so the tool shows IsSelectionItemPatternAvailable=true and all the method are invokable. If this is a TODO for this PR then lets' add TODO and create workitems for this. This change introduces an error to our UIA surface - we now expose something advertising ListItem but not implementing SelectionItem. Another point is that now we will have ListItem control types which are not children of List control type - this is a finer point but is also a violation of Windows accessibility standards.

2. Cool. But how it works in the virtual message list for example? Does it work correctly reflecting number of items before/after next batch is loaded?

3. Agreed. Let's create TODOs in code and workitems.

---
In reply to: [184240921](https://github.com/Microsoft/react-native-windows/pull/1803#discussion_r184240921) [](ancestors = 184240921,184237387,184221383,184140661)

Reply:
1. I checked what we get with this implementation (as in the PR) and the IsSelectionItemPatternAvailable returns 'false' so we are OK in regards not to advertise something that is not available. But implementing selection pattern is something to consider so I'll add TODO (and work item) for it. I didn't see a mandatory requirement to a have a List Control Type always when having List Item Control Type. It works fine without List Control
2. "one out of x" seems to be based on what is in the automation tree. I don't think it will work if you have batches / pages loaded unloaded but this is also questionable as intent e.g. if you have historical items for a year do you what to announce something like 1 of 10000? I would guess not and I don't think it was intended to do that.
3. I'll add TODOs to reflect these. 

---
In reply to: [184245460](https://github.com/Microsoft/react-native-windows/pull/1803#discussion_r184245460) [](ancestors = 184245460,184240921,184237387,184221383,184140661)

Reply:
I'm just concerned that by implementing partial support for ListItem control type we will just increase number of accessibility bugs reported since the partial implementation breaks many requirements and Microsoft standards for accessibility. But if you checked all the corner cases and will add TODO for remaining work I'm okay with this.

---
In reply to: [184539583](https://github.com/Microsoft/react-native-windows/pull/1803#discussion_r184539583) [](ancestors = 184539583,184245460,184240921,184237387,184221383,184140661)